client/a8c-for-agencies/sections/partner-directory/agency-details/index.tsx

```diff
@@ -38,9 +40,11 @@ const AgencyDetailsForm = ( { initialFormData }: Props ) => {
 
 	const { validate, validationError, updateValidationError } = useDetailsFormValidation();
 
+	const agency = useSelector( getActiveAgency );
+
 	const onSubmitSuccess = useCallback(
 		( response: Agency ) => {
-			response && reduxDispatch( setActiveAgency( response ) );
+			response && reduxDispatch( setActiveAgency( { ...agency, ...response } ) );
```

Comment:
We obtained the current agency data and merged it with the new agency data (response) to preserve the remaining unchanged agency data.